### PR TITLE
Fix border padding on the right and bottom sides not being placed (fix #3993)

### LINKED ISSUE #3993: Aseprite sometimes incorrectly handles paddings between sprites and whole sheet when packing sprite sheets
Aseprite sometimes has bugs when packing the sprite sheets, related to spacings between the sprites and the edges of the atlas.

For example, a free set of template animations by the author [Zeggy](https://zegley.itch.io), which can be downloaded from [itch.io](https://zegley.itch.io/2d-platformermetroidvania-asset-pack), Aseprite packs the sprite sheet incorrectly if you specify options:

- Sheet Type: Packed
- Trim Cels: True
- Border Padding: 1
- Spacing (Shape Padding): 1

image

In the picture below, I filled the entire area not used by the sprites with blue, and the places where the sprites are in contact with each other and with the borders of the atlas became visible.

ASEPRITE_BUG

### Aseprite and System version
Aseprite: v1.3-rc4-x64, installer (from Humble Bundle)
System:
- OS: Windows 10 Pro 21H2
- Processor: Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz   3.19 GHz
- RAM: 64,0 GB
- Graphics: NVIDIA GeForce GTX 760

### PR DESCRIPTION
**Pull request by Gasparoken**
Prior to this fix, border padding was clipped to the edge of sprites on the right and bottom in all export sprite sheet types.

This commit includes tests for 'Export Sprite Sheet' command using shape padding, border padding, inner padding with 'Packed' sheet type
+ 'Trim Cels' = true.

This fix complements the aseprite/laf commit named: "Fix overlapped shape padding between frame images on Packed sheet type"

fix #3993

### CODE REVIEW

File: src/app/doc_exporter.cpp [hunk 1]
```diff
@@ -1302,7 +1302,14 @@ void DocExporter::trimTexture(const Samples& samples,
         sample.isEmpty())
       continue;
 
-    bounds |= sample.inTextureBounds();
+    // We add the border padding in the sample size to do an union
+    // between full bounds and sample's inTextureBounds (it
+    // shouldn't make full bounds bigger).
+    gfx::Rect sampleBounds(sample.inTextureBounds());
+    sampleBounds.w += m_borderPadding;
+    sampleBounds.h += m_borderPadding;
```
Comment (martincapello) [inline src/app/doc_exporter.cpp:1310]: You can use:

```suggestion
    sampleBounds.inflate(m_borderPadding)
```

Or even avoid declaring a new variable:
```c++
bounds |= gfx::Rect(sample.inTextureBounds()).inflate(m_borderPadding);
```
Not sure if this is preferred or not...just mentioning the possibility.

File: tests/scripts/app_command.lua [hunk 2]
```diff
@@ -54,6 +54,53 @@ do -- ExportSpriteSheet
     11,8,11,21,11,8,11,
     11,8,11,21,11,11,8,
   })
+
+  local s = Sprite{ fromFile="sprites/4f-index-4x4.aseprite" }
+  app.command.ExportSpriteSheet {
+    type=SpriteSheetType.PACKED,
+    textureFilename="_test_export_spritesheet1.png",
```
Comment (martincapello) [inline tests/scripts/app_command.lua:61]: I think we should use a different name here? There is another test that uses this filename already. Maybe _test_export_spritesheet2.png?

**Review by martincapello — APPROVED** [review]
Just a couple of minor details.

**Comment by Gasparoken** [comment]
Ready for review

**Comment by Gasparoken** [comment]
Added tests will succeed if https://github.com/aseprite/laf/pull/64 on LAF is applied first.

**Comment by dacap** [comment]
Tested, works great! I've just cherry picked this into bcbe61c882c992291f14fc0b56464c120c40869d with the laf change https://github.com/aseprite/laf/commit/cbb157c9dc021bf8474e8d00a3145ae7a322528c

**Pull request closed without merging**